azurerm/resource_arm_container_registry.go

```diff
@@ -499,3 +560,39 @@ func validateAzureRMContainerRegistryName(v interface{}, k string) (warnings []s
 
 	return warnings, errors
 }
+
+func expandNetworkRuleSet(d *schema.ResourceData) *containerregistry.NetworkRuleSet {
+	configs := d.Get("network_access_profile").(*schema.Set).List()
+	config := configs[0].(map[string]interface{})
+
+	virtualNetworkRuleConfigs := config["subnet_rule"].([]interface{})
+	virtualNetworkRules := make([]containerregistry.VirtualNetworkRule, 0, len(virtualNetworkRuleConfigs))
+
+	for _, virtualNetworkRuleInterface := range virtualNetworkRuleConfigs {
+		config := virtualNetworkRuleInterface.(map[string]interface{})
+		virtualNetworkRules =
+			append(virtualNetworkRules, containerregistry.VirtualNetworkRule{
+				Action:                   containerregistry.Action(config["action"].(string)),
+				VirtualNetworkResourceID: utils.String(config["subnet_id"].(string)),
+			})
+	}
+
+	ipRuleConfigs := config["ip_rule"].([]interface{})
+	ipRules := make([]containerregistry.IPRule, 0, len(ipRuleConfigs))
```

Comment:
(as above) we can remove this final argument here:

```suggestion
	ipRules := make([]containerregistry.IPRule, 0)
```